packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,383 @@
+// @ts-check
+import { Buffer } from 'buffer';
+import ReadlineTransform from 'readline-transform';
+import { Readable } from 'stream';
+import { assert, Fail, q } from '@agoric/assert';
+import { createSHA256 } from './hasher.js';
+
+/**
+ * @typedef { import('./swingStore').SwingStoreExporter } SwingStoreExporter
+ *
+ * @typedef {{
+ *   initTranscript: (vatID: string) => void,
+ *   rolloverSpan: (vatID: string) => void,
+ *   getCurrentTranscriptBounds: (vatID: string) => { startPos: number, endPos: number, hash: string, size: number },
+ *   writeTranscriptItem: (vatID: string, item: string) => void,
+ *   readTranscript: (vatID: string, startPos?: number, endPos?: number) => IterableIterator<string>,
+ *   closeTranscript: (vatID: string) => void,
+ *   getExportRecords: (includeHistorical: boolean) => Iterable<[key: string, value: string]>,
+ *   getArtifactNames: (includeHistorical: boolean) => Iterable<string>,
+ *   getTranscript: (vatID: string, startPos: number) => AsyncIterable<Uint8Array>
+ *   importTranscript: (artifactName: string, exporter: SwingStoreExporter, artifactMetadata: Map) => Promise<void>,
+ * }} TranscriptStore
+ *
+ * @typedef {{
+ *   dumpTranscripts: (includeHistorical?: boolean) => any,
+ * }} TranscriptStoreDebug
+ *
+ */
+
+function* empty() {
+  // Yield nothing
+}
+
+/**
+ * @param {unknown} position
+ * @returns {asserts position is number}
+ */
+
+function insistTranscriptPosition(position) {
+  assert.typeof(position, 'number');
+  assert(position >= 0);
+}
+
+/**
+ * @param {*} db
+ * @param {() => void} ensureTxn
+ * @param {(key: string, value: string) => void} noteExport
+ * @returns { TranscriptStore & TranscriptStoreDebug }
+ */
+export function makeTranscriptStore(db, ensureTxn, noteExport) {
+  db.exec(`
+    CREATE TABLE IF NOT EXISTS transcriptItems (
+      vatID TEXT,
+      position INTEGER,
+      item TEXT,
+      PRIMARY KEY (vatID, position)
+    )
+  `);
+
+  db.exec(`
+    CREATE TABLE IF NOT EXISTS transcriptSpans (
+      vatID TEXT,
+      startPos INTEGER,
+      endPos INTEGER,
+      hash TEXT,
+      current INTEGER,
```

Comment:
If it does, let's use it, at least in the schema / `CREATE TABLE` definition. SQLite is pretty casual about types under the hood (it will happily store blobs and strings in a column marked INTEGER, and they round trip correctly), but it'd be nice for documentation purposes. That said, I'm not sure that `true` and `false` are legal values to send into the SQLite API, we might have to `Number()`/`Boolean()` them on the way in/out.

Reply:
Sqlite does not do BOOLEAN.  An INTEGER column with a value of 0 or 1 is the official, canonical way to do this.